### DISCUSSION TITLE
Failed to compile on Mac

### DISCUSSION THREAD

**Discussion by liangcl0928** [aws/aws-sdk-cpp#2501] [Q&A] [answered]
The error messages reported are:

aws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:241:20: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

aws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:248:24: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

aws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:252:24: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

aws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:260:24: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

ws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:265:24: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

aws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:270:24: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

aws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:276:20: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

aws-sdk-cpp/src/aws-cpp-sdk-core/source/net/linux-shared/SimpleUDP.cpp:282:20: Implicit conversion loses integer precision: 'ssize_t' (aka 'long') to 'int'

The build option "AWS_WARNINGS_ARE_ERRORS" is turned off, "Treat Warning as Errors" option of Build Settings of Xcode is set to "No", but still the project won't compile.

My environment:
Mac 13.2.1 with M2 Pro
gcc version:
Apple clang version 14.0.0 (clang-1400.0.29.202)
Target: arm64-apple-darwin22.3.0

**Comment by yasminetalby** [comment, marked as answer]
Hello @liangcl0928 ,

Thank you very much for your submission.
Could you please provide a little bit more context?
Are you experiencing this when initially building the project? 
Could you provide me with the steps that you have taken that have lead to this behaviors so I can better understand your usage of the SDK?

Thank you very much for your time and collaboration.

Sincerely,

Yasmine

**Reply by jmklix** [reply]
Closing for now because of no response. Please let us know if you are still running into any error when compiling on Mac
